rfc/parameterizedMetrics.md

```diff
@@ -0,0 +1,124 @@
+Parameterized Metric Support
+============================
+
+Parameterized metrics are a subset of dynamic request metrics.  Specifically they focus on modifying the calculation of metrics by altering values used in the calculation of existing metrics or mapping their values at aggregation time.
+
+The expected contract of parameterized metrics is as follows: Given a base metric with api name `metricName`, it can be suffixed with a procedure-call list of keyword parameter names and values.  These values will be used in the calculation of the resulting new metric.
+
+So
+```
+"metricName(parameterName=parameterValue)"
+```
+
+will result in a dynamic metric based on `metricName`, but substituting the parameter value when calculating it.  Also, the above string will be used as the column name in the resulting report response schema. 
+
+Motivation
+----------
+
+Standard metrics in Fili are configured at load time.  However if reporting requires many different similar caluclations, such as adjusting precision or converting values of different units into a shared aggregate, or applying custom metric filters, it may be impractical or even impossible to configure all the interesting report metrics.
+
+With parameterized metrics, open-ended variations may be supported for a given conceptual metric calculation, simplifying configuration and possibly unlocking otherwise impractical freedom in metric calculation.
+
+
+The structure of a parameterized metric:
+----------------------------------------
+
+```
+metricName(parameterName=value)
```

Comment:
We should make it clear here that this is considered the name of the metric, unless aliased, i.e. this exact same string needs to be used in the having clause.

Furthermore, is this allowed:

`metricName(parameterName=value,AS=metricName)`.

If it is, then we should sit down and think about the rules of aliasing. Consider:

`mertricName(parameterName=value,AS=metricName),metricName(paramaterName=value2,AS=metricName2)`.

Which `metricName` should the second metric expression refer to? The first expression, or the base metric?

Reply:
How soon an alias becomes available is an interesting implementation problem.

It would be easiest to say that aliasing simply won't work between metrics.